app/controllers/dashboards_controller.rb

```diff
@@ -7,6 +7,20 @@ class DashboardsController < ApplicationController
 
   def index
     @term = params[:term]
-    @results = SearchService.run(page: params[:page], term: @term)
+    search_and_count_results(params[:page])
+  end
+
+  private
+
+  def search_and_count_results(page)
+    if @term.present?
+      all_results = SearchService.run(term: @term)
```

Comment:
Hmm 🤔 

I don't believe this change has introduced what may be a performance issue, but I think we may have one.

It looks to me with this line that we are asking Rails to return all matching results from the database, then asking Kaminari to paginate them. I think when we come to use live data and someone puts in a term like `waste` or `limited` this may cause a very slow response, or even crash the page.

So I think we'll want to be doing something similar to what we did in the front-office when dealing with this issue https://github.com/DEFRA/waste-carriers-front-office/blob/master/app/controllers/dashboards_controller.rb#L7-L10

How that affects this is, if we want to show the count we may want to do that separately as a count only i.e. get ActiveRecord/Postgres to only return the count and not the data, rather than count the number of records returned.

Reply:
Good point. I'll have a crack at sorting it out!

Reply:
Actually, given that we were calling `Kaminari.paginate_array(combined_results).page(page)` previously anyway (just in the service instead) we may have had this issue already. Doesn't mean it's worth fixing now though.

Reply:
Looks like this would be a lot easier if we were on Rails 5 and could use `or`: https://stackoverflow.com/a/9540911

Reply:
Lets get this bad boy aligned then upgrade!